### PR TITLE
PD-1333 / 25.04 / Pd 1333 stig response for security site

### PR DESCRIPTION
**Pull request by micjohnson777**
This PR creates the first STIG compliance article for the General Operating System SRG response.

Applies to 24.10 in initial draft, with updates for 25.04 planned.

Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

File: content/Solutions/Optimizations/STIGCompliance.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+ - TrueNAS STIG Compliance
+---
+
+
+## TrueNAS Compliance
+TrueNAS falls into the category of an appliance with its own operating system as covered in [General Purpose Operating System SRG](https://www.stigviewer.com/stig/general_purpose_operating_system_srg/) findings.
+Through connection to Active Directory, TrueNAS also complies with the [Active Directory Domain Security Technical Implementation Guide SRG](https://www.stigviewer.com/stig/active_directory_domain/) findings related to authentication and access controls for user, group, and systems.
+
+## Customizing TrueNAS Security Options for STIG Compliance
+Many areas of compliance with the STIG SGR findings are automatically addressed through the TrueNAS kernel and middleware, but some are optional settings and features in the TrueNAS UI administration users customize to suit individual use cases and security policies.
```
Comment (billohanlon) [inline content/Solutions/Optimizations/STIGCompliance.md:20]: SGR --> SRG

File: content/Solutions/Optimizations/STIGCompliance.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+Installing TrueNAS on servers other than iXsystems-provided systems, should read and follow guidance in the [TrueNAS Hardware Guide]({{< relref "SCALEHardwareGuide.md" >}}).
+
+## Configuring TrueNAS
+After installing TrueNAS, users must complete the initial configuration of network, storage, users, sharing, and data backup solutions. See [Configuration Instructions]({{< relref "/SCALE/gettingstarted/configure/_index.md" >}}) for more information.
+
+### Administration Accounts
+TrueNAS creates the root user and an administration user at installation.
+Some releases of TrueNAS might only have **root** as the default administration user, while other releases have either the **admin** or **truenas_admin** as the default user.
+
+TrueNAS systems with the **root** user, or either the **admin** or **truenas_admin** user should create a new administration user with full control privileges and assign a complex password that you should change on a frequent basis.
```
Comment (billohanlon) [inline content/Solutions/Optimizations/STIGCompliance.md:42]: The government standards for passwords just changed (within the last couple weeks.)  They don't suggest changing passwords frequently any longer.  Might want to change that to something like "assign a complex password and follow current password guidelines for managing it" or something.

File: content/Solutions/Optimizations/STIGCompliance.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+|----------|-------|---------|
+| **Audit** | **Audit Service Backend Failed** | Set alert level preference to send notifications when an auditing function failure occurs to promptly correct the issue and not lose audit logs. |
+|  | **Audit Service Setup Failed** | Set alert level preference to send notifications when the auditing setup fails to correct the issue promptly and not lose audit logs. |
+| **Certificates** | <li>**Certificate is Expiring Soon**<br><li>**Certificate is Expiring**<br><li>**Certificate has Expired**</li>  | Set alert level preferences to send notifications when a certificate is about to or has expired to either renew or replace the certificate before functions relying on certificates are impacted, and to keep those functions protected. |
+|  | **Certificate Revoked** | Set alert preferences to send notifications when a certificate is revoked to promptly address the issue or obtain a new certificate. |
+|  | **WEeb UI HTTPS Certificate Setup Failed** | Set alert level preferences to send notifications when the web UI HTTPS certificate setup fails to promptly address issues that impact the security of HTTPS access to the TrueNAS web UI. |
```
Comment (billohanlon) [inline content/Solutions/Optimizations/STIGCompliance.md:176]: WEeb --> Web

File: content/Solutions/Optimizations/STIGCompliance.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+Select the checkbox(es) for releases you want to delete from the list of inactive releases.
+Maintaining releases does not pose security risks but does consume space on the boot pool.
+See [Boot Pool Management]({{< relref "ManageBootEnvironSCALE.md" >}}) for more information on working with boot pool environments.
+
+### Virtualization
+This document does not cover the virtual machine envirnoments created by users.
+STIG compliance for these users-deployed environments is based on the operating system and applications deployed in these VMs.
+
+## Future STIG Compliance
+TrueNAS does not comply with STIG for the following findings that are planned for a future release:
```
Comment (billohanlon) [inline content/Solutions/Optimizations/STIGCompliance.md:263]: Should probably call out the specific STIG that we don't comply with here (GPOS STIG). 
(Not complying with STIGs in general isn't really a thing.)

File: content/Solutions/Optimizations/STIGCompliance.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
+---
```
Comment (billohanlon) [inline content/Solutions/Optimizations/STIGCompliance.md:1]: Should there be a reference to the FIPS support in this document?  Do we have the FIPS stuff documented yet?

File: content/Solutions/Optimizations/STIGCompliance.md (hunk 1 above)
Comment (micjohnson777) [reply content/Solutions/Optimizations/STIGCompliance.md:20]: oops... fixed

File: content/Solutions/Optimizations/STIGCompliance.md (hunk 2 above)
Comment (micjohnson777) [reply content/Solutions/Optimizations/STIGCompliance.md:42]: change made in next commit

File: content/Solutions/Optimizations/STIGCompliance.md (hunk 3 above)
Comment (micjohnson777) [reply content/Solutions/Optimizations/STIGCompliance.md:176]: oops... fixed

File: content/Solutions/Optimizations/STIGCompliance.md (hunk 4 above)
Comment (micjohnson777) [reply content/Solutions/Optimizations/STIGCompliance.md:263]: added particulars here in next commit.

File: content/Solutions/Optimizations/STIGCompliance.md (hunk 5 above)
Comment (micjohnson777) [reply content/Solutions/Optimizations/STIGCompliance.md:1]: We mention FIPS, or FIPS compliance in some of the articles but don't have one article about it generally. I'll add FIPS to this in the next commit

**Review by billohanlon — DISMISSED** [review]
LGTM

**Review by tonyriv3 — APPROVED** [review]

**Pull request merged**